heat/core/_operations.py

```diff
@@ -163,23 +168,24 @@ def __get_out_params(target, other=None, map=None):
     if out is not None:
         sanitation.sanitize_out(out, output_shape, output_split, output_device, output_comm)
         t1, t2 = sanitation.sanitize_distribution(t1, t2, target=out)
-        out.larray[:] = operation(
-            t1.larray.type(promoted_type), t2.larray.type(promoted_type), **fn_kwargs
+    else:
+        out_tensor = torch.empty(output_shape, dtype=promoted_type)
```

Comment:
2 comments here:
- `output_shape` is the global (memory-distributed) shape of the output DNDarray, here you're initializing a potentially huge torch tensor. In this case you should call 
```
factories.empty(output_shape, dtype=..., split=..., device=...)
```
and that will take care of only initializing slices of the global array on each process

- if I understand the numpy docs correctly, this empty `out` DNDarray only needs to be initialized if `where` is not None.